src/hotspot/share/opto/lcm.cpp

```diff
@@ -518,7 +518,8 @@ Node* PhaseCFG::select(
   uint score   = 0; // Bigger is better
   int idx = -1;     // Index in worklist
   int cand_cnt = 0; // Candidate count
-  bool block_size_threshold_ok = (block->number_of_nodes() > 10) ? true : false;
+  bool block_size_threshold_ok = recalc_pressure_nodes != NULL &&
+                                 (block->number_of_nodes() > 10) ? true : false;
```

Comment:
And this really should be:
` block_size_threshold_ok = (recalc_pressure_nodes != NULL) && (block->number_of_nodes() > 10);`